snapshot/snapshot.sh

```diff
@@ -0,0 +1,353 @@
+#!/bin/bash
+
+# Copyright 2022 Kontain
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[ "$TRACE" ] && set -x
+
+KONTAIN_NAMESPACE="kontain-snapshot-ns"
```

Comment:
Seems like we should add something to this to ensure uniqueness. 
Maybe append "-$$".
If there are multiple users working on the same cluster, we need to be sure they don't collide with each other.

Reply:
This does not need to be unique except for this cluster and just during this process. It gets removed after snapshot process completed.

Reply:
but if we have 2 o more deployments $$ will allow concurrent run, also have to  do that to all temporary files. So will do that

Reply:
The image directory ($DIR/image) will also suffer from the same problem.

Reply:
added a generated id to all names and directories that are used